crates/router/src/connector/bluesnap/transformers.rs

```diff
@@ -116,10 +127,82 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for BluesnapPaymentsRequest {
             payment_method,
             currency: item.request.currency,
             card_transaction_type: auth_mode,
+            three_d_secure: None,
+        })
+    }
+}
+
+impl TryFrom<&types::PaymentsCompleteAuthorizeRouterData> for BluesnapPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsCompleteAuthorizeRouterData) -> Result<Self, Self::Error> {
+        let redirection_response: BluesnapRedirectionResponse = item
+            .request
+            .payload
+            .clone()
+            .ok_or(errors::ConnectorError::MissingRequiredField {
```

Comment:
add a new error for `MissingPayloadField{fieldname}`, MissingRequiredField is reserved for customer missing an required field. this is not customer input.